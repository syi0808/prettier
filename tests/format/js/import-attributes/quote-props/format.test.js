const errors = {
  acorn: ["quoted-keys.js"],
  espree: ["quoted-keys.js"],
};
runFormatTest(import.meta, ["babel", "typescript"], { errors });
runFormatTest(import.meta, ["babel", "typescript"], {
  quoteProps: "consistent",
  errors,
});
runFormatTest(import.meta, ["babel", "typescript"], {
  quoteProps: "preserve",
  errors,
});
