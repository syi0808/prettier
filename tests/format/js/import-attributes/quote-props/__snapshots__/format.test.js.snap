// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quoted-keys.js [acorn] format 1`] = `
"Unexpected token (1:29)
> 1 | import json from './mod.js' with { "type":"json" };
    |                             ^
  2 | import json from './mod.js' with { "foobar":"json" };
  3 | import json from './mod.js' with { "type-foo" :"json" };
  4 | import json from './mod.js' with { "333foo" :"json" };
Cause: Unexpected token (1:28)"
`;

exports[`quoted-keys.js [espree] format 1`] = `
"Unexpected token with (1:29)
> 1 | import json from './mod.js' with { "type":"json" };
    |                             ^
  2 | import json from './mod.js' with { "foobar":"json" };
  3 | import json from './mod.js' with { "type-foo" :"json" };
  4 | import json from './mod.js' with { "333foo" :"json" };
Cause: Unexpected token with"
`;

exports[`quoted-keys.js - {"quoteProps":"consistent"} [acorn] format 1`] = `
"Unexpected token (1:29)
> 1 | import json from './mod.js' with { "type":"json" };
    |                             ^
  2 | import json from './mod.js' with { "foobar":"json" };
  3 | import json from './mod.js' with { "type-foo" :"json" };
  4 | import json from './mod.js' with { "333foo" :"json" };
Cause: Unexpected token (1:28)"
`;

exports[`quoted-keys.js - {"quoteProps":"consistent"} [espree] format 1`] = `
"Unexpected token with (1:29)
> 1 | import json from './mod.js' with { "type":"json" };
    |                             ^
  2 | import json from './mod.js' with { "foobar":"json" };
  3 | import json from './mod.js' with { "type-foo" :"json" };
  4 | import json from './mod.js' with { "333foo" :"json" };
Cause: Unexpected token with"
`;

exports[`quoted-keys.js - {"quoteProps":"consistent"} format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
quoteProps: "consistent"
                                                                                | printWidth
=====================================input======================================
import json from './mod.js' with { "type":"json" };
import json from './mod.js' with { "foobar":"json" };
import json from './mod.js' with { "type-foo" :"json" };
import json from './mod.js' with { "333foo" :"json" };
import json from './mod.js' with { "0" :"json" };
import json from './mod.js' with { "0n" :"json" };
import json from './mod.js' with { "33333" :"json" };
import json from './mod.js' with { "33333n" :"json" };
import json from './mod.js' with { "33333b" :"json" };
import json from './mod.js' with { "03333" :"json" };

=====================================output=====================================
import json from "./mod.js" with { type: "json" };
import json from "./mod.js" with { foobar: "json" };
import json from "./mod.js" with { "type-foo": "json" };
import json from "./mod.js" with { "333foo": "json" };
import json from "./mod.js" with { "0": "json" };
import json from "./mod.js" with { "0n": "json" };
import json from "./mod.js" with { "33333": "json" };
import json from "./mod.js" with { "33333n": "json" };
import json from "./mod.js" with { "33333b": "json" };
import json from "./mod.js" with { "03333": "json" };

================================================================================
`;

exports[`quoted-keys.js - {"quoteProps":"preserve"} [acorn] format 1`] = `
"Unexpected token (1:29)
> 1 | import json from './mod.js' with { "type":"json" };
    |                             ^
  2 | import json from './mod.js' with { "foobar":"json" };
  3 | import json from './mod.js' with { "type-foo" :"json" };
  4 | import json from './mod.js' with { "333foo" :"json" };
Cause: Unexpected token (1:28)"
`;

exports[`quoted-keys.js - {"quoteProps":"preserve"} [espree] format 1`] = `
"Unexpected token with (1:29)
> 1 | import json from './mod.js' with { "type":"json" };
    |                             ^
  2 | import json from './mod.js' with { "foobar":"json" };
  3 | import json from './mod.js' with { "type-foo" :"json" };
  4 | import json from './mod.js' with { "333foo" :"json" };
Cause: Unexpected token with"
`;

exports[`quoted-keys.js - {"quoteProps":"preserve"} format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
quoteProps: "preserve"
                                                                                | printWidth
=====================================input======================================
import json from './mod.js' with { "type":"json" };
import json from './mod.js' with { "foobar":"json" };
import json from './mod.js' with { "type-foo" :"json" };
import json from './mod.js' with { "333foo" :"json" };
import json from './mod.js' with { "0" :"json" };
import json from './mod.js' with { "0n" :"json" };
import json from './mod.js' with { "33333" :"json" };
import json from './mod.js' with { "33333n" :"json" };
import json from './mod.js' with { "33333b" :"json" };
import json from './mod.js' with { "03333" :"json" };

=====================================output=====================================
import json from "./mod.js" with { "type": "json" };
import json from "./mod.js" with { "foobar": "json" };
import json from "./mod.js" with { "type-foo": "json" };
import json from "./mod.js" with { "333foo": "json" };
import json from "./mod.js" with { "0": "json" };
import json from "./mod.js" with { "0n": "json" };
import json from "./mod.js" with { "33333": "json" };
import json from "./mod.js" with { "33333n": "json" };
import json from "./mod.js" with { "33333b": "json" };
import json from "./mod.js" with { "03333": "json" };

================================================================================
`;

exports[`quoted-keys.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import json from './mod.js' with { "type":"json" };
import json from './mod.js' with { "foobar":"json" };
import json from './mod.js' with { "type-foo" :"json" };
import json from './mod.js' with { "333foo" :"json" };
import json from './mod.js' with { "0" :"json" };
import json from './mod.js' with { "0n" :"json" };
import json from './mod.js' with { "33333" :"json" };
import json from './mod.js' with { "33333n" :"json" };
import json from './mod.js' with { "33333b" :"json" };
import json from './mod.js' with { "03333" :"json" };

=====================================output=====================================
import json from "./mod.js" with { type: "json" };
import json from "./mod.js" with { foobar: "json" };
import json from "./mod.js" with { "type-foo": "json" };
import json from "./mod.js" with { "333foo": "json" };
import json from "./mod.js" with { "0": "json" };
import json from "./mod.js" with { "0n": "json" };
import json from "./mod.js" with { "33333": "json" };
import json from "./mod.js" with { "33333n": "json" };
import json from "./mod.js" with { "33333b": "json" };
import json from "./mod.js" with { "03333": "json" };

================================================================================
`;
